src/main/perl/ComponentProxyList.pm

```diff
@@ -58,55 +58,55 @@ sub reportComponents
     return SUCCESS;
 }
 
-# Run the pre-config $hook, possibly timing out after $timeout seconds
+
+# common code for pre/post hooks
+# Run the pre/post-config $hook, possibly timing out after $timeout seconds
+# If data is defined, it is passed via stdin in JSON format
 # Returns 1 on success, 0 on failure
-sub pre_config_actions
+# TODO: the types are probably not preserved all the way here,
+#   so the JSON format e.g. might have strings for integers
+sub _config_actions
```

Comment:
Oh - I see why this was necessary now. I know last time you had to split a commit it was super-painful but in this case it's just one file so should be relatively easy? I definitely think it would be better as two commits.